### DISCUSSION TITLE
How can I define context/gender for translation?

### DISCUSSION THREAD

**Discussion by Cochonours** [vinissimus/next-translate#977] [Q&A] [answered]
I need to translate strings differently depending on the gender and overall context of the variable. I see that in other, similar libs, it is done through a special context variable, e.g.:

`t("go_to_country", { context: country_code, name: country })`

So the translation files can do something like:

```
"go_to_country": "Aller en {{name}}",
"go_to_country_uk": "Aller au Roayume-Uni",
"go_to_country_mc": "Aller à Monaco",
...
```

So similar to "count" special notation. It doesn't work with next-translate and I don't see anything in the docs. Is it possible?

**Comment by aralroca** [comment]
Context is not implemented. Please confirm if you can solve using this way https://github.com/aralroca/next-translate/issues/611#issuecomment-840404206 One of the goals of the library is to keep tiny (~1kb), so we are only supporting the really necessary features. If you have some troubles with the solution that I proposed and you think that this feature is really necessary to implement, feel free to open an issue to report it to implement.

**Comment by Cochonours** [comment, marked as answer]
Thanks for your quick answer! I actually was thinking about doing that before checking this thread back, and the solution you linked to works, but only partially: the problem is that I would need to exhaustively define a translation for each possible context value.

The perfect "working workaround" is do add a fallback, so in my case:

`      { t("go_to_country_" + country_code.toLowerCase(), { name: country }, { fallback: 'go_to_country' }) }`

This works like a charm!

**Reply by aralroca** [reply]
Cool 😊 Thanks for sharing!
